pvlib/data/gold/sdm.py

```diff
@@ -0,0 +1,847 @@
+                        I_o_ref=device.I_o_ref,
+                        R_sh_ref=device.R_sh_ref,
+                        R_s=device.R_s,
+                        EgRef=EgRef,
+                        dEgdT=dEgdT)
+                    # More convienent variable names
+                    i_l, i_0, r_s, r_sh, nNsVth = device_params
+                    # Ideal, regular, or degraded series resistance
+                    r_s = r_s_scale * r_s
+                    # Ideal, regular, or degraded shunt conductance
+                    # json spec excludes IEEE 754 inf, so use shunt
+                    #  conductance instead of shunt resistance
+                    g_sh = g_sh_scale / r_sh
+                    # Reverse engineer the diode voltage range
+                    i_sc = pvlib.pvsystem.i_from_v(r_sh, r_s, nNsVth, 0., i_0,
+                                                   i_l)  # Not a gold value
+                    v_oc = pvlib.pvsystem.v_from_i(r_sh, r_s, nNsVth, 0., i_0,
+                                                   i_l)  # Not a gold value
+                    # For min diode voltage, go slightly less than Isc, even
+                    #  when r_s==0
+                    v_d_min = i_sc * r_s - 0.02 * v_oc
+                    # For max diode voltage, go slightly greater than Voc
+                    v_d_max = 1.01 * v_oc
+                    # More I-V points towards Voc
+                    v_d = v_d_min + \
+                        (v_d_max - v_d_min) * \
+                        numpy.log10(numpy.linspace(1., 10.**1.01, NUM_PTS))
+                    # Bishop's method is explicit and inherently vectorized
+                    i_gold = i_l - i_0 * numpy.expm1(v_d / nNsVth) - g_sh * v_d
+                    v_gold = v_d - r_s * i_gold
+
+                    # Record lossless, machine-interoperable hex values for
+                    #  the (V, I) pairs
+                    device_dict["iv_curves"].append({
+                        "G": G.hex(),
+                        "T": T.hex(),
+                        "i_l": i_l.hex(),
+                        "i_0": i_0.hex(),
+                        "r_s": r_s.hex(),
+                        "g_sh": g_sh.hex(),
+                        "nNsVth": nNsVth.hex(),
+                        "v_gold": [v.hex() for v in v_gold.tolist()],
+                        "i_gold": [i.hex() for i in i_gold.tolist()]
+                    })
+
+                    # Solve the current sum at diode node residual as a
+                    #  numerically reliable interval
+                    current_sum_residual_interval_list = \
+                        [interval(i_l) - interval(i_0) * imath.expm1(
+                            (interval(v) + interval(i)*interval(r_s)) /
+                            interval(nNsVth)) - interval(g_sh) *
+                            (interval(v) + interval(i)*interval(r_s)) -
+                            interval(i) for v, i in zip(v_gold, i_gold)]
+
```

Comment:
What's the advantage of using intervals here, where all the intervals are single numbers?